cli/src/lib.rs

```diff
@@ -24,40 +25,72 @@ use {
 #[derive(Parser, Debug)]
 #[clap(name = "gluesql", about, version)]
 struct Args {
-    /// sled-storage path to load
-    #[clap(short, long, value_parser)]
-    path: Option<PathBuf>,
-
     /// SQL file to execute
     #[clap(short, long, value_parser)]
     execute: Option<PathBuf>,
 
     /// PATH to dump whole database
     #[clap(short, long, value_parser)]
     dump: Option<PathBuf>,
+
+    /// Storage type to store data, default is memory
+    #[clap(short, long, value_parser)]
+    storage: Option<Storage>,
+
+    /// Storage path to load
+    #[clap(short, long, value_parser)]
+    path: Option<PathBuf>,
+}
+
+#[derive(clap::ValueEnum, Debug, Clone)]
+enum Storage {
+    Memory,
+    Sled,
+    Json,
 }
 
 pub fn run() -> Result<()> {
     let args = Args::parse();
+    let (path, storage) = match (args.path, args.storage) {
```

Comment:
It looks that two separate `match` statements can be merged into a single one.

```rust
let path = args
    .path
    .as_ref()
    .map(PathBuf::as_path)
    .and_then(Path::to_str);

match (path, args.storage) {
    (None, None) | (None, Some(Storage::Memory)) => {
        println!("[memory-storage] initialized");

        run(MemoryStorage::default(), args.execute);
    }
    (Some(_), Some(Storage::Memory)) => {
        panic!("failed to load memory-storage: it should be without path");
    }
    (Some(path), Some(Storage::Sled)) => {
        println!("[sled-storage] connected to {}", path);

        run(
            SledStorage::new(path).expect("failed to load sled-storage"),
            args.execute,
        );

        if let Some(dump_path) = args.dump {
            let mut storage = SledStorage::new(path).expect("failed to load sled-storage");
            dump_database(&mut storage, dump_path)?;
        }
    }
    (Some(path), Some(Storage::Json)) => {
        println!("[json-storage] connected to {}", path);

        run(
            JsonStorage::new(path).expect("failed to load json-storage"),
            args.execute,
        );
    }
    (None, Some(_)) | (Some(_), None) => {
        panic!("both path and storage should be specified");
    }
}
```

Reply:
Thanks!

I realized current logic does not make dump_database works so that i used `match with 3 elements` including `args.dump`